keychain_win.cpp

```diff
@@ -156,9 +156,21 @@ void updateError(keychain::Error &err) {
  * The result is wrapped in a ScopedLpwstr.
  */
 ScopedLpwstr makeTargetName(const std::string &package,
-                            const std::string &service,
-                            const std::string &user) {
-    return utf8ToWideChar(package + "." + service + '/' + user);
+                            const std::string &service, const std::string &user,
+                            keychain::Error &err) {
+    const auto result = utf8ToWideChar(package + "." + service + '/' + user);
+    if (!result) {
+        updateError(err);
+
+        // make really sure that we set an error code of we will return nullptr
+        if (!err) {
+            err.type = keychain::ErrorType::GenericError;
+            err.message = "failed to create credential target name";
```

Comment:
While I'm not sure about all error messages created by the OSs, the ones I've seen and the ones we produce are sentences.
```suggestion
            err.message = "Failed to create credential target name.";
```